fdbserver/worker.actor.cpp

```diff
@@ -448,6 +452,28 @@ ACTOR Future<Void> runProfiler(ProfilerRequest req) {
 	}
 }
 
+void runHeapProfiler() {
+#if defined(__linux__) && defined(USE_GPERFTOOLS) && !defined(VALGRIND)
+	if (IsHeapProfilerRunning()) {
+		HeapProfilerDump("User triggered heap dump");
+	} else {
+		TraceEvent("ProfilerError").detail("Message", "HeapProfiler Unsupported");
```

Comment:
These should probably be two different messages to not cause confusion?

```suggestion
		TraceEvent("ProfilerError").detail("Message", "HeapProfiler already running");
```

Reply:
Good point. Actually the profiler should be started before taking the dump.